### ISSUE TITLE
gumbel_softmax during update

### ISSUE THREAD

**Issue by ChengshuLi** [shariqiqbal2810/maddpg-pytorch#34]
Hi, thanks for this great repo!

Quick question, should this line be `hard=False` instead? Otherwise, how does the gradient flow through?

https://github.com/shariqiqbal2810/maddpg-pytorch/blob/40388d7c18e4662cf23c826d97e209df9003d86c/algorithms/maddpg.py#L143

Thanks a lot for your help!!

**Comment by shariqiqbal2810** [comment]
Hi! The `hard` flag specifies whether the action "appears" to be one-hot, but it always acts like a gumbel-softmax sample in the backward pass. You can look at the implementation of that function for more details.

**Comment by ChengshuLi** [comment]
Got it. Thank you!
